src/tribler/core/components/ipv8/eva/protocol.py

```diff
@@ -152,31 +179,103 @@ def send_binary(self, peer: Peer, info: bytes, data: bytes) -> Awaitable[Transfe
                 It is limited by several GB, but the protocol is slow by design, so
                 try to send less rather than more.
         """
+        if self.shutting_down:
+            raise TransferException('The protocol is shutting down')
+
         if not data:
             raise ValueException('The empty data binary passed')
 
         if peer == self.community.my_peer:
             raise ValueException('The receiver can not be equal to the sender')
 
         nonce = self.random.randint(0, MAX_U32)
-        transfer = OutgoingTransfer(data, self.outgoing, peer, info, nonce, self.settings,
-                                    on_complete=self.on_send_complete, on_error=self.on_error)
+        data_size = len(data)
+        if data_size > self.settings.binary_size_limit:
+            raise SizeException(f'Data size limit {self.settings.binary_size_limit} has been exceeded: {data_size}')
+
+        transfer = OutgoingTransfer(
+            container=self.outgoing,
+            peer=peer,
+            info=info,
+            data=data,
+            data_size=data_size,
+            nonce=nonce,
+            settings=self.settings,
+            parent_task_group=self.task_group,
```

Comment:
Should be renamed to `protocol_task_group` here and in other places:
```suggestion
            protocol_task_group=self.task_group,
```

Reply:
Renamed